File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

public class Solution {

  static ArrayList<String> dictionary = new ArrayList<>();
  static String[] array = {"A", "E", "I", "O", "U"};

  public int solution(String word) {
    int answer = 0;

    dfs(0, "");

    answer = dictionary.indexOf(word);

    return answer;
  }

  private void dfs(int depth, String str) {
    dictionary.add(str);

    if(depth == 5) {
      return;
    }

    for(int i = 0; i < 5; i++) {
      dfs(depth + 1, str + array[i]);
    }
  }

}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
value = -1

def solution(word):
    answer = 0
    dic = {}
    word_list = ['A', 'E', 'I', 'O', 'U']
    recursive("", dic, word_list)
    
    answer = dic[word]
    
    return answer

def recursive(text, dic, word_list):
    global value
    if len(text) >= 6:
        return
    
    value += 1
    dic[text] = value
    
    for w in word_list:
        recursive(text + w, dic, word_list)
    